bundles/org.openhab.binding.sunsa/src/main/java/org/openhab/binding/sunsa/internal/device/SunsaDeviceHandler.java

```diff
@@ -0,0 +1,188 @@
+        });
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateState();
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_CONFIGURABLE_POSITION: {
+                if (command instanceof StopMoveType) {
+                    // We don't support this
+                    LOGGER.warn("Received {} command which is not supported and is explicitly ignored.", command);
+                    return;
+                }
+
+                final int rawPosition = getPositionAdapter(channelUID.getId())
+                        .getRawPosition(getPositionInterpolationFactor(command));
+                LOGGER.debug("Attempting update to raw position to: {}", rawPosition);
+                updatePosition(rawPosition);
+                break;
+            }
+
+            case CHANNEL_RAW_POSITION: {
+                final int rawPosition = ((DecimalType) command).intValue();
+                LOGGER.debug("Attempting update to raw position to: {}", rawPosition);
+                updatePosition(rawPosition);
+                break;
+            }
+
+            case CHANNEL_BATTERY_LEVEL: {
+                updateState();
+            }
+        }
+    }
+
+    public PositionAdapter getPositionAdapter(final String channelID) {
+        switch (channelID) {
+            case CHANNEL_CONFIGURABLE_POSITION:
+                return PositionAdapters.configurablePositionAdapter(config.configurablePositionClosed,
+                        config.configurablePositionOpen);
+            default:
+                throw new IllegalArgumentException("Unknown channel ID: " + channelID);
+        }
+    }
+
+    /**
+     * Return a value between [0.0, 1.0] for the value of the command.
+     */
+    private float getPositionInterpolationFactor(final Command command) {
+        final float rawValue;
+        if (command instanceof PercentType) {
+            rawValue = ((PercentType) command).floatValue();
+        } else if (command instanceof UpDownType) {
+            rawValue = ((UpDownType) command).as(PercentType.class).floatValue();
+        } else if (command instanceof OpenClosedType) {
+            rawValue = ((OpenClosedType) command).as(PercentType.class).floatValue();
+        } else if (command instanceof DecimalType) {
+            rawValue = ((DecimalType) command).floatValue();
+        } else {
+            throw new IllegalArgumentException("Unknown command: " + command);
+        }
+
+        return rawValue / PercentType.HUNDRED.floatValue();
+    }
+
+    private void updatePosition(final int rawPosition) {
+        final int updatedRawPosition;
+        try {
+            updatedRawPosition = sunsaService.setDevicePosition(config.id, rawPosition);
+        } catch (final SunsaException e) {
+            LOGGER.error("Could not update position for device: {}", config.id, e);
```

Comment:
The logging should be removed. All info is allready present at the updateStatus. 
Should the device really be set to offline? Is there no possible way the value is rejected, while the cloud/device is perfectly healthy?